src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -32,21 +34,20 @@ public async Task InvokeAsync(HttpContext context)
 			var transaction = _tracer.StartTransactionInternal($"{context.Request.Method} {context.Request.Path}",
 				ApiConstants.TypeRequest);
 
-			transaction.Context.Request = new Request
+			var url = new Url
+			{
+				Full = context.Request?.Path.Value,
+				HostName = context.Request.Host.Host,
+				Protocol = GetProtocolName(context.Request.Protocol),
+				Raw = context.Request?.Path.Value //TODO
+			};
+			transaction.Context.Request = new Payload.Request(context.Request.Method, url)

```

Comment:
In internal code we can still directly set things on `Context`. This may be a little bit confusing, but from perf reasons I think we should do it this way - we avoid copying stuff this way.